test-fips-docker.sh

```diff
@@ -0,0 +1,463 @@
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[1;33m'
+NC='\033[0m' # No Color
+
+# Function to print colored output
+print_status() {
+    local status=$1
+    local message=$2
+    if [ "$status" = "SUCCESS" ]; then
+        echo -e "${GREEN}✓${NC} $message"
+    elif [ "$status" = "WARNING" ]; then
+        echo -e "${YELLOW}⚠${NC} $message"
+    else
+        echo -e "${RED}✗${NC} $message"
+    fi
+}
+
+# Function to show usage
+show_help() {
+    cat << EOF
+Usage: $0 [OPTIONS]
+
+FIPS Testing Script for GeoServer using Podman
+
+OPTIONS:
+    -h, --help          Show this help message
+    -c, --clean         Clean up existing containers and images before starting
+    -p, --port PORT     Use custom port (default: 8080)
+    -s, --skip-build    Skip building GeoServer WAR file (use existing)
+    -d, --skip-download Skip downloading Jetty (use existing)
+
+EXAMPLES:
+    $0                    # Run with default settings
+    $0 --clean           # Clean up and run
+    $0 --port 8081       # Use port 8081 instead of 8080
+    $0 --skip-build      # Skip building GeoServer (faster if already built)
+
+PREREQUISITES:
+    - Docker installed and running
+    - Maven (mvn) available in PATH
+    - wget available in PATH
+    - curl available in PATH
+    - Sufficient disk space for GeoServer build (~2GB)
+    - Sufficient memory for Maven build (~2GB RAM)
+
+EOF
+}
+
+# Parse command line arguments
+CLEAN_MODE=false
+CUSTOM_PORT=8080
+SKIP_BUILD=false
+SKIP_DOWNLOAD=false
+
+while [[ $# -gt 0 ]]; do
+    case $1 in
+        -h|--help)
+            show_help
+            exit 0
+            ;;
+        -c|--clean)
+            CLEAN_MODE=true
+            shift
+            ;;
+        -p|--port)
+            CUSTOM_PORT="$2"
+            shift 2
+            ;;
+        -s|--skip-build)
+            SKIP_BUILD=true
+            shift
+            ;;
+        -d|--skip-download)
+            SKIP_DOWNLOAD=true
+            shift
+            ;;
+        *)
+            echo "Unknown option: $1"
+            show_help
+            exit 1
+            ;;
+    esac
+done
+
+# Set FIPS environment variables
+GEOSERVER_KEYSTORE_TYPE=BCFKS
+GEOSERVER_KEYSTORE_PROVIDER=BC
+
+# Find available port if the specified port is in use
+REQUESTED_PORT=$CUSTOM_PORT
+CUSTOM_PORT=$(find_available_port $CUSTOM_PORT)
+if [ $? -ne 0 ]; then
+    print_status "ERROR" "Failed to find available port starting from $REQUESTED_PORT"
+    exit 1
+fi
+
+if [ "$REQUESTED_PORT" != "$CUSTOM_PORT" ]; then
+    print_status "WARNING" "Port $REQUESTED_PORT is in use, using port $CUSTOM_PORT instead"
+fi
+
+echo "=== GeoServer FIPS Testing Script (Docker) ==="
+echo "Port: $CUSTOM_PORT"
+echo "Clean mode: $CLEAN_MODE"
+echo "Skip build: $SKIP_BUILD"
+echo "Skip download: $SKIP_DOWNLOAD"
+echo
+
+# Function to cleanup on exit
+cleanup() {
+    echo
+    echo "Cleaning up..."
+    # Remove temporary files if they exist
+    rm -f jetty-distribution-9.4.44.v20210927.tar.gz
```

Comment:
jetty version to the latest release. can also find this in ngs-geoserver versions.env